### ISSUE TITLE
Python: Do we always want the most updated version of requirements? Or just some?

### ISSUE THREAD

**Issue by joeflack4** [monarch-initiative/mondo-ingest#57]
Summary:

**Comment by matentzn** [comment]
I dont have any better idea to be honest. I would still prefer using fixed locked dependencies so that the code does not suddenly break during a future run where some dependency introduced a breaking change though

**Comment by joeflack4** [comment]
I think that makes a lot of sense.

My only worry is that I know myself, and I know I'll run into bugs and my first reaction will not always be to check if I have the latest version of the dependency installed.

Basically that is what's going to end up being what drives our versioning-up of dependencies in `mondo-ingest`. Encountering bugs, finding out that these libs are outdated, and then updating them.

It would be good to know when a new version is released, but even then, I wouldn't have time to fully evaluate. And it's really hard to tell if a new version might have a bug / breaking change in it without using it first.

---

Perhaps the best thing to do would be, if the exception is being raised from one of these dependencies, e.g. `oaklib` or `sssom`, to then check to see if there's a new version, and print out a message about that. But that's probably too much work.

I suppose I'll come to a decision on this on my own and then come back and close the issue.

**Comment by matentzn** [comment]
Yeah, not clear cut for me either.. Hopefully these libraries stablise at some point. Mayeb better to just add the commands into the makefile so at least when you are hacking the makefile you see which versions are being installed..
